services/galley/src/Galley/API/Teams.hs

```diff
@@ -136,8 +141,9 @@ uncheckedDeleteTeam zusr zcon tid = do
         let e = newEvent TeamDelete tid now
         let r = list1 (userRecipient zusr) (membersToRecipients (Just zusr) membs)
         pushSome ((newPush1 zusr (TeamEvent e) r & pushConn .~ zcon) : events)
+        when ((view teamBinding . Data.tdTeam <$> team) == Just Binding) $
+            deleteUsers (map (view userId) membs)
```

Comment:
> Perhaps mark team as deleted before deleting users?

Good point. I actually wanted to allow this to be repeatable in the way you mentioned and your suggestion makes sense to me.